sympy/matrices/matrices.py

```diff
@@ -1103,69 +1106,80 @@ def diagonalize(self, reals_only=False, sort=False, normalize=False):
+        if error_when_incomplete:
+            if flags.get('multiple'):
+                eigs_count = len(eigs)
+            else:
+                eigs_count = sum(m for m in eigs.values())
+            if eigs_count != self.cols:
+                raise MatrixError("Could not compute eigenvalues for {}".format(self))
 
         return eigs
 
-    def eigenvects(self, error_when_incomplete=True, **flags):
-        """Return list of triples (eigenval, multiplicity, basis).
 
-        The flag ``simplify`` has two effects:
-            1) if bool(simplify) is True, as_content_primitive()
-            will be used to tidy up normalization artifacts;
-            2) if nullspace needs simplification to compute the
-            basis, the simplify flag will be passed on to the
-            nullspace routine which will interpret it there.
+    def eigenvects(self, error_when_incomplete=True, simplify=None,
+                   chop=False, **flags):
+        """Returns a list of triples (eigenvalue, algebraic multiplicity,
+        eigenspace basis).
 
         Parameters
         ==========
 
-        error_when_incomplete : bool
+        error_when_incomplete : bool, optional (default: True)
             Raise an error when not all eigenvalues are computed. This is
             caused by ``roots`` not returning a full list of eigenvalues.
 
-        If the matrix contains any Floats, they will be changed to Rationals
-        for computation purposes, but the answers will be returned after being
-        evaluated with evalf. If it is desired to removed small imaginary
-        portions during the evalf step, pass a value for the ``chop`` flag.
-        """
-        from sympy.matrices import eye
+        simplify : bool or None, optional (default: None)
+            If None, the only simplification done is to a nullspace, when
+            its basis cannot be computed otherwise.
+            If True, then the eigenvectors will be simplified as well.
+            The value of False disables all simplification.
 
-        simplify = flags.get('simplify', True)
-        if not isinstance(simplify, FunctionType):
-            simpfunc = _simplify if simplify else lambda x: x
-        primitive = flags.get('simplify', False)
-        chop = flags.pop('chop', False)
+        chop : bool, optional (default: False)
+            Truncate approximation errors caused by the conversion of
+            Floats to Rationals in the process of computing eigenvalues
+            and eigenvectors.
 
-        flags.pop('multiple', None)  # remove this if it's there
+        flags : optional keyword arguments to be passed to ``roots``.
+
+        """
+        # For this method, returning a list is not possible.
+        if flags.get('multiple'):
+            raise NotImplementedError(filldedent('''The multiple flag is not
+                supported since eigenvectors cannot be returned as a list.'''))
```

Comment:
I'm not sure this error really makes sense.

Reply:
Do you mean the message (which can probably be improved) or the fact of raising an error? I don't see any way for multiple=True to be implemented by eigenvects. We can't just return a list of tuples (eigenvalue, eigenvector), repeating some eigenvalues due to multiplicity. For one thing, the number of linearly independent eigenvectors (geometric multiplicity) may be less than the algebraic multiplicity of the eigenvalue.